buildSrc/src/main/kotlin/source-sets-conventions.gradle.kts

```diff
@@ -124,8 +124,12 @@ tasks.withType(KotlinCompilationTask::class).configureEach {
             allWarningsAsErrors = true
         }
         if (overriddenLanguageVersion != null) {
-            languageVersion = KotlinVersion.fromVersion(overriddenLanguageVersion!!)
-            freeCompilerArgs.add("-Xsuppress-version-warnings")
+            kotlin {
```

Comment:
It is strange to configure extension within the task. I would propose to add `compilerOptions` DSL upper in the script where `kotlin {` extension is being configured (line 27):
 ```
kotlin {
    compilerOptions {
           // These configuration replaces 'languageSettings' config on line 67
           progressiveMode.set(true)
           optIn.addAll(
                 listOf(
                    "kotlin.ExperimentalMultiplatform",
                    "kotlin.ExperimentalSubclassOptIn",
                    "kotlinx.serialization.InternalSerializationApi",
                    "kotlinx.serialization.SealedSerializationApi",
                 )
            )
           if (overriddenLanguageVersion != null) {
               languageVersion = KotlinVersion.fromVersion(overriddenLanguageVersion!!)
               freeCompilerArgs.add("-Xsuppress-version-warnings")
           }
     }
     ...
```